asset-transfer-basic/rest-api-typescript/src/assets.router.ts

```diff
@@ -0,0 +1,348 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This sample is intended to work with the basic asset transfer
+ * chaincode which imposes some constraints on what is possible here.
+ *
+ * For example,
+ *  - There is no validation for Asset IDs
+ *  - There are no error codes from the chaincode
+ *
+ * To avoid timeouts, long running tasks should be decoupled from HTTP request
+ * processing
+ *
+ * Submit transactions can potentially be very long running, especially if the
+ * transaction fails and needs to be retried one or more times
+ *
+ * To allow requests to respond quickly enough, this sample queues submit
+ * requests for processing asynchronously and immediately returns 202 Accepted
+ */
+
+import express, { Request, Response } from 'express';
+import { body, validationResult } from 'express-validator';
+import { Contract } from 'fabric-network';
+import { getReasonPhrase, StatusCodes } from 'http-status-codes';
+import { Queue } from 'bullmq';
+import { AssetNotFoundError } from './errors';
+import { evatuateTransaction } from './fabric';
+import { addSubmitTransactionJob } from './jobs';
+import { logger } from './logger';
+
+const { ACCEPTED, BAD_REQUEST, INTERNAL_SERVER_ERROR, NOT_FOUND, OK } =
+  StatusCodes;
+
+export const assetsRouter = express.Router();
+
+assetsRouter.get('/', async (req: Request, res: Response) => {
+  logger.debug('Get all assets request received');
+
+  try {
+    const mspId = req.user as string;
+    const contract = req.app.get(mspId).assetContract as Contract;
+
+    const data = await evatuateTransaction(contract, 'GetAllAssets');
+    let assets = [];
+    if (data.length > 0) {
+      assets = JSON.parse(data.toString());
+    }
+
+    return res.status(OK).json(assets);
+  } catch (err) {
+    logger.error({ err }, 'Error processing get all assets request');
+    return res.status(INTERNAL_SERVER_ERROR).json({
+      status: getReasonPhrase(INTERNAL_SERVER_ERROR),
+      timestamp: new Date().toISOString(),
+    });
+  }
+});
+
+assetsRouter.post(
+  '/',
+  body().isObject().withMessage('body must contain an asset object'),
+  body('id', 'must be a string').notEmpty(),
+  body('color', 'must be a string').notEmpty(),
+  body('size', 'must be a number').isNumeric(),
+  body('owner', 'must be a string').notEmpty(),
+  body('appraisedValue', 'must be a number').isNumeric(),
+  async (req: Request, res: Response) => {
+    logger.debug(req.body, 'Create asset request received');
+
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(BAD_REQUEST).json({
+        status: getReasonPhrase(BAD_REQUEST),
+        reason: 'VALIDATION_ERROR',
+        message: 'Invalid request body',
+        timestamp: new Date().toISOString(),
+        errors: errors.array(),
+      });
+    }
+
+    const mspId = req.user as string;
```

Comment:
mspid is for the org. isn't req.user an individual user?

Reply:
There's currently an API for each org, rather than an individual user